ports/espressif/common-hal/wifi/Radio.c

```diff
@@ -242,6 +242,10 @@ void common_hal_wifi_radio_start_ap(wifi_radio_obj_t *self, uint8_t *ssid, size_
     esp_wifi_set_config(WIFI_IF_AP, config);
 }
 
+bool common_hal_wifi_radio_get_ap_active(wifi_radio_obj_t *self) {
+    return self->ap_mode;
```

Comment:
I'll have to dig deeper in the code / my old notes, but I think `mode` and `enabled` are independent. i.e., it's possible to be in AP mode, but wifi is disabled, and therefore the AP is not "active" (isn't sending beacons and can't be connected to).

Reply:
I'll look into that. The check for `.connected` is clearer, because it checks whether the interface is up. But the interface check is false when the AP is on, and there doesn't seem to be an equivalent call in the API to check that. If you know of one, let me know. I spent some time looking.

One thing I noted is that the code calls a number of `esp_` routines without checking their return codes. That's something to address in the future, perhaps.

Reply:
`.enabled` doesn't actually enable/disable the wifi on CYW43 yet:
https://github.com/adafruit/circuitpython/blob/e8711ee836eba9248cf4a73cd3e06ecf91d3e15c/ports/raspberrypi/common-hal/wifi/Radio.c#L75-L77

Reply:
I don't know the espressif NETIF API very well, but quick test confirms `.connected` is `True` (only) when: mode is STA or AP+STA, wifi is enabled, station is started, and station is connected (has an IP address). The event handler essentially defines connected as having an IP address for espressif, but NETIF up seems to be a good alternate / proxy.

I'm not as familiar with CYW43 / raspberrypi. Since the API is a subset, I use it in pretty basic ways and haven't done any development for it. Been intending to test all the permutations, but haven't gotten to it.

But maybe we can either use NETIF up with the AP NETIF (e.g., `return self->ap_mode && esp_netif_is_netif_up(self->ap_netif);`, or (like station uses IP address) use some piece of AP info that's only valid when the AP is up, like `ipv4_address_ap`:

```py
>>> wifi.radio.start_ap("Bob", "YourUncle")
>>> wifi.radio.ap_active
True
>>> wifi.radio.enabled = False
>>> wifi.radio.ap_active  # should be False
True
>>> wifi.radio.ipv4_address_ap
>>> wifi.radio.enabled = True
>>> wifi.radio.ipv4_address_ap
192.168.4.1
>>> wifi.radio.stop_ap()
>>> wifi.radio.ipv4_address_ap
>>> wifi.radio.ap_active
False
```

> One thing I noted is that the code calls a number of esp_ routines without checking their return codes

For the espressif AP, that's probably my fault.

Reply:
I think `return self->ap_mode && esp_netif_is_netif_up(self->ap_netif);` will work:
https://github.com/adafruit/circuitpython/blob/e8711ee836eba9248cf4a73cd3e06ecf91d3e15c/ports/espressif/common-hal/wifi/Radio.c#L433

Reply:
Thanks -- I did that -- I didn't realize there was a `self->ap_netif`. I first just checked `self->started`, but that is better.

Reply:
I think AP mode + started/enabled covers all the cases, but it's predicated on all of the current logic (which, who knows, maybe we change to reconcile espressif API & CYW API). NETIF up should hopefully be definitive always. I'll test the artifacts more thoroughly this weekend.